### ISSUE TITLE
Change default configuration folder in registry

### ISSUE THREAD

**Issue by LeonarddeR** [nvaccess/nvda#11129]
# Is your feature request related to a problem? Please describe.
Currently, it is possible to override NVDA"s configuration folder using the "-c" parameter. This however does not work is NVDA is started from ease of access using the automatically start NVDA after logon checkbox in the general settings

### Describe the solution you'd like
Allow setting a config parameter in the user registry to override the default. Any config path on the command line should still override. Also, this config parameter shouldn't apply to portable copies.

### Additional context
This allows people to sett the default configuration to a path that's under version control, a users home folder mapped in active directory or in a cloud sync folder (dropbox, OneDrive, etc.)

**Comment by lukaszgo1** [comment]
I've been thinking about something similar for some time now however I would implement it differently and extend the implementation to other command line parameters.
To give some real-life examples:
- As an employer I don't want my employees to install and use add-ons, so I'd like to always   pass --disable-addons to the NVDA executable.
- As a user who is worried about my privacy I want to always disable logging (choosing disabled from the GUI does not disable it completely)
- As a developer for debugging purposes I would like to always redirect my logs to some other location.


I propose the following implementation:
When starting NVDA  reads command line params from NVDA_params.ini. This file can be placed in the user config folder and in the NVDA install location. The parameters from NVDA install locaytion have priority over what is in the user folder - they cannot be overwritten from the command line. Commends are appreciated.

**Comment by leonardder** [comment]
> When starting NVDA reads command line params from NVDA_params.ini. This file can be placed in the user config folder and in the NVDA install location. The parameters from NVDA install locaytion have priority over what is in the user folder - they cannot be overwritten from the command line. 

Interesting idea. [ConfigArgParse](https://pypi.org/project/ConfigArgParse/) should be able to do this, though I'm not sure whether you can set inheritance order in there. It gives us a new ini parcer, may be an implementation based on configObj is nicer for us. Overall, I like the approach.

**Comment by DrSooom** [comment]
See also: https://github.com/nvaccess/nvda/issues/9284#issuecomment-464391861

**Comment by seanbudd** [comment]
this should be considered as a part of #16599, however a different approach than a reg key might be better
